lib/updatehub/updatehub_priv.h

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2018 O.S.Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ *
+ *  @brief This file contains structures representing JSON messages
+ *  exchanged with a UpdateHub
+ */
+
+#ifndef __UPDATEHUB_PRIV_H__
+#define __UPDATEHUB_PRIV_H__
+
+#define UPDATEHUB_API_HEADER \
+	"Api-Content-Type: application/vnd.updatehub-v1+json"
+
+enum updatehub_uri_path {
+	UPDATEHUB_PROBE = 0,
+	UPDATEHUB_REPORT,
+	UPDATEHUB_DOWNLOAD,
+};
+
+enum updatehub_state {
+	UPDATEHUB_STATE_DOWNLOADING = 0,
+	UPDATEHUB_STATE_DOWNLOADED,
+	UPDATEHUB_STATE_INSTALLING,
+	UPDATEHUB_STATE_INSTALLED,
+	UPDATEHUB_STATE_REBOOTING,
+	UPDATEHUB_STATE_ERROR,
+};
+
+static char *updatehub_response(enum updatehub_response response)
+{
+	switch (response) {
+	case UPDATEHUB_NETWORKING_ERROR:
+		return "Fail to connect to the UpdateHub server";
+	case UPDATEHUB_INCOMPATIBLE_HARDWARE:
+		return "Incompatible hardware";
+	case UPDATEHUB_METADATA_ERROR:
+		return "Fail to parse or to encode the metadata";
+	case UPDATEHUB_DOWNLOAD_ERROR:
+		return "Fail while downloading the update package";
+	case UPDATEHUB_INSTALL_ERROR:
+		return "Fail while installing the update package";
+	case UPDATEHUB_FLASH_INIT_ERROR:
+		return "Fail to initilialize the flash";
+	case UPDATEHUB_NO_UPDATE:
+		return "No update available";
+	default:
+		return NULL;
+	}
+}
+
+static const char *uri_path(enum updatehub_uri_path type)
+{
+	switch (type) {
+	case UPDATEHUB_PROBE:
+		return "upgrades";
+	case UPDATEHUB_REPORT:
+		return "report";
+	case UPDATEHUB_DOWNLOAD:
+		return "products";
+	default:
+		return NULL;
+	}
+}
+
+static const char *state_name(enum updatehub_state state)
+{
+	switch (state) {
+	case UPDATEHUB_STATE_DOWNLOADING:
+		return "downloading";
+	case UPDATEHUB_STATE_DOWNLOADED:
+		return "downloaded";
+	case UPDATEHUB_STATE_INSTALLING:
+		return "installing";
+	case UPDATEHUB_STATE_INSTALLED:
+		return "installed";
+	case UPDATEHUB_STATE_REBOOTING:
+		return "rebooting";
+	case UPDATEHUB_STATE_ERROR:
+		return "error";
+	default:
+		return NULL;
+	}
+}
+
+struct resp_probe_objects {
+	const char *mode;
+	const char *sha256sum;
+	int size;
+};
+
+struct resp_probe_objects_array {
+	struct resp_probe_objects objects;
+};
+
+struct resp_probe_any_boards {
+	struct resp_probe_objects_array objects[2];
+	size_t objects_len;
+	const char *product;
+	const char *supported_hardware;
+};
+
+struct resp_probe_some_boards {
+	struct resp_probe_objects_array objects[2];
+	size_t objects_len;
+	const char *product;
+	const char *supported_hardware[40];
```

Comment:
Instead of magic constant, please create a define that describes the value better. Why value 40 here, could the value be configured via Kconfig if end user wants to minimize the amount of memory allocated?